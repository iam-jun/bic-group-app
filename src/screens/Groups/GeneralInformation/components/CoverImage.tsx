import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React, {useState} from 'react';
import ButtonWrapper from '~/beinComponents/Button/ButtonWrapper';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import {scaleCoverHeight} from '~/theme/dimension';
import {useKeySelector} from '~/hooks/selector';
import groupsKeySelector from '~/screens/Groups/redux/keySelector';
import Image from '~/beinComponents/Image';
import Text from '~/beinComponents/Text';
import images from '~/resources/images';
import spacing from '~/theme/spacing';

interface Props {
  testID?: string;
  onEditCover: () => void;
  backgroundUrl: string;
  canEditInfo: boolean;
}

const CoverImage = ({
  testID,
  onEditCover,
  backgroundUrl,
  canEditInfo,
}: Props) => {
  const [coverHeight, setCoverHeight] = useState<number>(210);
  const theme = useTheme() as ExtendedTheme;
  const {colors} = theme;
  const styles = themeStyles(theme, coverHeight);
  const loadingCover = useKeySelector(groupsKeySelector.loadingCover);

  const onCoverLayout = (e: any) => {
    if (!e?.nativeEvent?.layout?.width) return;
    const coverWidth = e.nativeEvent.layout.width;
    const coverHeight = scaleCoverHeight(coverWidth);
    setCoverHeight(coverHeight);
  };
  const textColor = !loadingCover ? colors.purple60 : colors.gray40;

  return (
    <View testID={testID}>
      <View style={styles.coverHeader}>
        <Text.H5 color={colors.neutral80} useI18n>
          settings:title_cover
        </Text.H5>
        {!!canEditInfo && (
          <ButtonWrapper
            testID="cover.button_edit"
            onPress={onEditCover}
            disabled={loadingCover}>
            <Text.H6 testID="cover.text_edit" color={textColor} useI18n>
              settings:title_edit
            </Text.H6>
          </ButtonWrapper>
        )}
      </View>
      <View onLayout={onCoverLayout}>
        {!loadingCover ? (
          <Image
            testID="cover.image"
            style={styles.cover}
            source={backgroundUrl || images.img_cover_default}
          />
        ) : (
          <View
            testID="cover.loading"
            style={[styles.cover, styles.imageLoading]}>
            <ActivityIndicator />
          </View>
        )}
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme, coverHeight: number) => {
  const {colors} = theme;

  return StyleSheet.create({
    coverHeader: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginHorizontal: spacing.margin.large,
      marginVertical: spacing.margin.small,
    },
    imageLoading: {
      backgroundColor: colors.gray20,
      justifyContent: 'center',
    },
    cover: {
      width: '100%',
      height: coverHeight,
    },
    basicInfoList: {
      marginHorizontal: spacing.margin.tiny,
    },
  });
};

export default CoverImage;
