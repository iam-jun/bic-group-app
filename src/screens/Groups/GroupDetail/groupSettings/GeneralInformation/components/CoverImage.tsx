import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React, {useState} from 'react';
import ButtonWrapper from '~/beinComponents/Button/ButtonWrapper';
import {useTheme} from 'react-native-paper';
import {ITheme} from '~/theme/interfaces';
import {scaleCoverHeight} from '~/theme/dimension';
import {useKeySelector} from '~/hooks/selector';
import groupsKeySelector from '~/screens/Groups/redux/keySelector';
import Image from '~/beinComponents/Image';
import Text from '~/beinComponents/Text';
import images from '~/resources/images';

interface Props {
  onEditCover: () => void;
}

const CoverImage = ({onEditCover}: Props) => {
  const [coverHeight, setCoverHeight] = useState<number>(210);
  const theme = useTheme() as ITheme;
  const {colors} = theme;
  const styles = themeStyles(theme, coverHeight);
  const loadingCover = useKeySelector(groupsKeySelector.loadingCover);
  const {background_img_url} =
    useKeySelector(groupsKeySelector.groupDetail.group) || {};

  const onCoverLayout = (e: any) => {
    if (!e?.nativeEvent?.layout?.width) return;
    const coverWidth = e.nativeEvent.layout.width;
    const coverHeight = scaleCoverHeight(coverWidth);
    setCoverHeight(coverHeight);
  };
  const textColor = !loadingCover ? colors.primary7 : colors.textDisabled;

  return (
    <View>
      <View style={styles.coverHeader}>
        <Text.H5 color={colors.iconTint} useI18n>
          settings:title_cover
        </Text.H5>
        <ButtonWrapper
          testID="cover.button_edit"
          onPress={onEditCover}
          disabled={loadingCover}>
          <Text.H6 testID="cover.text_edit" color={textColor} useI18n>
            settings:title_edit
          </Text.H6>
        </ButtonWrapper>
      </View>
      <View onLayout={onCoverLayout}>
        {!loadingCover ? (
          <Image
            testID="cover.image"
            style={styles.cover}
            source={background_img_url || images.img_cover_default}
          />
        ) : (
          <View
            testID="cover.loading"
            style={[styles.cover, styles.imageLoading]}>
            <ActivityIndicator />
          </View>
        )}
      </View>
    </View>
  );
};

const themeStyles = (theme: ITheme, coverHeight: number) => {
  const {spacing, colors} = theme;

  return StyleSheet.create({
    coverHeader: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginHorizontal: spacing.margin.large,
      marginVertical: spacing.margin.small,
    },
    imageLoading: {
      backgroundColor: colors.bgDisable,
      justifyContent: 'center',
    },
    cover: {
      width: '100%',
      height: coverHeight,
    },
    basicInfoList: {
      marginHorizontal: spacing.margin.tiny,
    },
  });
};

export default CoverImage;
