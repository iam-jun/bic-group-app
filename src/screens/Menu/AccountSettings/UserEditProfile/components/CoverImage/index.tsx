import {
  ActivityIndicator, StyleProp, StyleSheet, View, ViewStyle,
} from 'react-native';
import React from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import Text from '~/baseComponents/Text';
import Image from '~/beinComponents/Image';
import ButtonWrapper from '~/baseComponents/Button/ButtonWrapper';
import { spacing } from '~/theme';
import images from '~/resources/images';
import { scaleCoverHeight } from '~/theme/dimension';

interface Props {
  style?: StyleProp<ViewStyle>;
  url: string;
  loading: boolean;
  onPressEdit: () => void
}

const CoverImage = ({
  style, url, loading, onPressEdit,
}: Props) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = createStyles(theme);

  return (
    <View style={style}>
      <View style={styles.headerItem}>
        <Text.BodyM color={colors.neutral80} useI18n>
          settings:title_cover
        </Text.BodyM>
        <ButtonWrapper onPress={onPressEdit} disabled={loading}>
          <Text.H6
            testID="cover_image.edit"
            color={!loading ? colors.neutral80 : colors.gray40}
            style={styles.editBtn}
            useI18n
          >
            settings:title_edit
          </Text.H6>
        </ButtonWrapper>
      </View>
      <View
        style={{ paddingHorizontal: spacing.padding.large }}
        testID="cover_image.image"
      >
        {!loading ? (
          <Image
            style={styles.cover}
            source={url || images.img_cover_default}
          />
        ) : (
          <View style={[styles.cover, styles.imageLoading]}>
            <ActivityIndicator />
          </View>
        )}
      </View>
    </View>
  );
};

const createStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    headerItem: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingRight: spacing.padding.base,
      paddingVertical: spacing.padding.small,
      paddingLeft: spacing.padding.large,
      alignItems: 'center',
    },
    imageLoading: {
      backgroundColor: colors.gray20,
      justifyContent: 'center',
    },
    editBtn: { padding: spacing.padding.small },
    cover: {
      width: '100%',
      height: scaleCoverHeight(),
      borderRadius: spacing.borderRadius.small,
    },
  });
};

export default CoverImage;
