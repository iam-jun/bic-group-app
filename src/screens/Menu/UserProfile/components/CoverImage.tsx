import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { useState } from 'react';
import { StyleSheet, View } from 'react-native';
import Image from '~/beinComponents/Image';
import images from '~/resources/images';
import { spacing } from '~/theme';
import { scaleCoverHeight } from '~/theme/dimension';
import EditButton from './EditButton';

interface Props {
    id: string;
    userId: string;
    currentUsername: string;
    bgImg: any;
    coverHeight: number;
    onEdit: () => void;
}

const CoverImage = ({
  id, bgImg, userId, currentUsername, coverHeight, onEdit,
}: Props) => {
  const theme:ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <View testID="user_profile.cover_image">
      <Image
        style={[styles.cover, { height: (coverHeight * 2) / 3 }]}
        source={bgImg || images.img_cover_default}
      />
      <EditButton
        testID="user_profile.edit.cover_image"
        style={styles.editCoverPhoto}
        userId={userId}
        currentUsername={currentUsername}
        onPress={onEdit}
      />
    </View>
  )
}

const themeStyles = (
  theme: ExtendedTheme,
) => {
  const { colors } = theme;

  return StyleSheet.create({
    cover: {
      width: '100%',
    },
    editCoverPhoto: {
      position: 'absolute',
      top: spacing.margin.small,
      right: spacing?.margin.small,
    },
  });
}

export default CoverImage
