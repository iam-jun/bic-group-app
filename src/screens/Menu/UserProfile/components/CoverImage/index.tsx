import React from 'react';
import { StyleSheet, View } from 'react-native';
import Image from '~/components/Image';
import images from '~/resources/images';
import { dimension, spacing } from '~/theme';
import EditButton from '../EditButton';

interface Props {
  isCurrentUser: boolean;
  bgImg: any;
  coverHeight: number;
  onEdit: () => void;
}

const CoverImage = ({
  bgImg, isCurrentUser, coverHeight, onEdit,
}: Props) => (
  <View testID="user_profile.cover_image">
    <Image
      width={dimension.deviceWidth}
      style={[styles.cover, { height: (coverHeight * 2) / 3 }]}
      source={bgImg || images.img_cover_default}
    />
    <EditButton
      testID="user_profile.cover_image.edit_button"
      style={styles.editCoverPhoto}
      isCurrentUser={isCurrentUser}
      onPress={onEdit}
      icon="CameraSolid"
    />
  </View>
);

const styles = StyleSheet.create({
  cover: {
    width: '100%',
  },
  editCoverPhoto: {
    position: 'absolute',
    top: spacing.margin.large,
    right: spacing.margin.large,
  },
});

export default CoverImage;
