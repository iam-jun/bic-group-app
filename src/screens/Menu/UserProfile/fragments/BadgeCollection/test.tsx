import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
