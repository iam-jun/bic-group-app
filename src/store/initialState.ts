import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
