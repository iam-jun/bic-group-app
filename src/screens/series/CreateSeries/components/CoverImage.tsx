import React, { useState } from 'react';
import { StyleSheet, View } from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Text from '~/baseComponents/Text';
import spacing from '~/theme/spacing';
import UploadingImage from '~/beinComponents/UploadingImage';
import { uploadTypes } from '~/configs/resourceConfig';
import { IFilePicked } from '~/interfaces/common';
import dimension, { scaleSize } from '~/theme/dimension';
import Icon from '~/baseComponents/Icon';
import { Button } from '~/baseComponents';
import { checkPermission, permissionTypes } from '~/utils/permission';
import ImagePicker from '~/beinComponents/ImagePicker';
import { IArticleCover } from '~/interfaces/IPost';
import { IGetFile } from '~/store/uploader';

const COVER_WIDTH = dimension.deviceWidth;
const COVER_HEIGHT = scaleSize(200);

interface Props {
  style?: any;
  disabled?: boolean;
  coverMedia: IArticleCover;
  onUploadSuccess: (cover: IArticleCover) => void;
}

const CoverImage = ({
  style, coverMedia, disabled, onUploadSuccess,
}: Props) => {
  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  const [selectingCover, setSelectingCover] = useState<IFilePicked>();
  const [error, setError] = useState('');

  const onPressSelect = () => {
    checkPermission(
      permissionTypes.photo, (canOpenPicker) => {
        if (canOpenPicker) {
          ImagePicker.openPickerSingle({
            mediaType: 'photo',
          }).then((file) => {
            setSelectingCover(file);
          });
        }
      },
    );
  };

  const handleUploadSuccess = (file: IGetFile) => {
    if (file?.result?.id && file?.result?.url) {
      onUploadSuccess(file.result as IArticleCover);
    } else {
      console.error('\x1b[36m🐣️ index onUploadSuccess invalid result\x1b[0m');
    }
  };

  const onError = (error) => setError(error);

  const renderError = (error: string) => (
    <View style={[styles.cover, styles.errorContainer]}>
      <Icon icon="Image" size={30} tintColor={theme.colors.neutral40} />
      <Text.SubtitleM style={styles.textAddCover} useI18n>common:text_add_cover</Text.SubtitleM>
      <Text.BodyXS style={styles.textError}>{error}</Text.BodyXS>
      <Button onPress={onPressSelect}>
        <Text.LinkS useI18n>article:text_select_cover</Text.LinkS>
      </Button>
    </View>
  );

  const renderButton = () => {
    if (!coverMedia?.url && !error) {
      return (
        <Button
          testID="create_series.cover_image.button_add"
          style={[styles.centerButtonContainer, styles.emptyCoverContainer]}
          onPress={onPressSelect}
        >
          <Icon icon="Image" size={30} tintColor={theme.colors.neutral40} />
          <Text.SubtitleM style={styles.textAddCover} useI18n>common:text_add_cover</Text.SubtitleM>
        </Button>
      );
    }
    if (error) {
      return null;
    }
    return (
      <View style={styles.centerButtonContainer}>
        <Button.Neutral
          disabled={disabled}
          type="ghost"
          size="medium"
          onPress={onPressSelect}
          useI18n
        >
          series:text_update_cover
        </Button.Neutral>
      </View>
    );
  };

  return (
    <View style={style}>
      <UploadingImage
        uploadType={uploadTypes.postImage}
        style={styles.cover}
        file={selectingCover}
        fileName={selectingCover?.filename}
        url={!selectingCover && coverMedia.url}
        width={COVER_WIDTH}
        height={COVER_HEIGHT}
        onUploadSuccess={handleUploadSuccess}
        onError={onError}
        renderError={renderError}
      />
      {renderButton()}
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      paddingHorizontal: spacing?.padding.large,
      paddingVertical: spacing?.padding.small,
      alignItems: 'center',
      borderBottomColor: colors.neutral5,
      borderBottomWidth: 1,
    },
    textPostTo: {
    },
    cover: {
      width: '100%',
      height: scaleSize(200),
    },
    emptyCoverContainer: {
      backgroundColor: colors.neutral2,
    },
    textAddCover: {
      marginTop: spacing.margin.base,
    },
    iconCamera: {
      position: 'absolute',
      top: spacing.margin.small,
      right: spacing.margin.small,
    },
    errorContainer: {
      backgroundColor: colors.neutral2,
      justifyContent: 'center',
      alignItems: 'center',
    },
    textError: {
      color: colors.red40,
      paddingVertical: spacing.padding.small,
    },
    centerButtonContainer: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      justifyContent: 'center',
      alignItems: 'center',
    },
  });
};

export default CoverImage;
