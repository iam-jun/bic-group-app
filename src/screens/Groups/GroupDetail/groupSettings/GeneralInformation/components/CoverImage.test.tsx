import React from 'react';
import {StyleSheet} from 'react-native';
import initialState from '~/store/initialState';
import {fireEvent, renderWithRedux, configureStore} from '~/test/testUtils';
import {colors} from '~/theme';
import CoverImage from './CoverImage';

describe('CoverImage component', () => {
  const baseProps = {
    onEditCover: jest.fn(),
  };

  const mockStore = configureStore([]);

  const storeData = {...initialState};

  it('renders correctly', () => {
    const store = mockStore(storeData);
    const rendered = renderWithRedux(
      <CoverImage {...baseProps} />,
      store,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it(`should show loading`, () => {
    storeData.groups.loadingCover = true;
    const store = mockStore(storeData);

    const rendered = renderWithRedux(<CoverImage {...baseProps} />, store);

    const loadingComponent = rendered.getByTestId('cover.loading');
    expect(loadingComponent).not.toBeNull();
  });

  it(`should show cover`, () => {
    storeData.groups.loadingCover = false;
    const store = mockStore(storeData);

    const rendered = renderWithRedux(<CoverImage {...baseProps} />, store);

    const imageComponent = rendered.getByTestId('cover.image');
    const loadingComponent = rendered.findByTestId('cover.loading');
    expect(loadingComponent).not.toBe({});
    expect(imageComponent).toBeDefined();
  });

  it(`should disable button when loading`, () => {
    storeData.groups.loadingCover = true;
    const store = mockStore(storeData);

    const rendered = renderWithRedux(<CoverImage {...baseProps} />, store);

    const buttonComponent = rendered.getByTestId('cover.button_edit');
    const textComponent = rendered.getByTestId('cover.text_edit');
    const flattenedStyle = StyleSheet.flatten(textComponent.props.style);
    expect(flattenedStyle.color).toBe(colors.light.colors.textDisabled);
    expect(buttonComponent.props.accessibilityState.disabled).toBe(true);
  });

  it(`should call onEditCover when edit button press`, () => {
    storeData.groups.loadingCover = false;
    const store = mockStore(storeData);

    const onEditCover = jest.fn();

    const rendered = renderWithRedux(
      <CoverImage onEditCover={onEditCover} />,
      store,
    );

    const buttonComponent = rendered.getByTestId('cover.button_edit');
    fireEvent.press(buttonComponent);
    expect(onEditCover).toBeCalled();
  });
});
