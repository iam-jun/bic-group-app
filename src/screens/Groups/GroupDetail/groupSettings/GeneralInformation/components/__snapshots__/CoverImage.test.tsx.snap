// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImage component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 16,
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2F3945",
            "fontFamily": "OpenSans-SemiBold",
            "fontSize": 16,
          },
          Object {
            "color": "#2F3945",
          },
          undefined,
        ]
      }
    >
      Cover
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="cover.button_edit"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="button_wrapper.content"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2F3945",
                "fontFamily": "OpenSans-SemiBold",
                "fontSize": 14,
              },
              Object {
                "color": "#421187",
              },
              undefined,
            ]
          }
          testID="cover.text_edit"
        >
          Edit
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "position": "relative",
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f9f9f9",
              "height": 210,
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="Image__placeholder"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 210,
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          onFastImageError={[Function]}
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../assets/images/cover_default.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="cover.image"
        />
      </View>
    </View>
  </View>
</View>
`;
